2021-12-27-getting-started-with-purenix.md

```diff
@@ -0,0 +1,217 @@
+```
+
+Except for the pattern matching in `fromMaybe`, this should look pretty
+straight-forward.  You can see the PureScript string `greeting` becomes a
+simple Nix string.  `identity` has become a Nix function.  The constructors for
+`Maybe` (`Nothing` and `Just`) have become Nix functions[^2].  `fromMaybe` has
+become a Nix function.  If you work through the `__patternX` continuations in
+`fromMaybe`, you can see how they
+[emulate the pattern matching](https://jonascarpay.com/posts/2021-11-08-nix-adts.html)
+in the PureScript `fromMaybe` function.
+
+Calling functions written in PureScript from Nix is quite fun.  It feels quite
+similar to working in the Haskell or PureScript REPL.  Here is an example of
+using the above functions from the Nix REPL:
+
+```console
+$ nix repl ./output/Main/default.nix
+nix-repl> identity greeting
+"Hello, world!"
+nix-repl> fromMaybe "bye" (Just "hello")
+"hello"
+nix-repl> fromMaybe "bye" Nothing
+"bye"
+```
+
+Now that you have a taste of PureNix, the next section explains how to get
+started actually writing PureNix code.
+
+## Getting Started
+
+There is a post on the NixOS Discourse that explains how to
+[put together a development environment for PureNix](https://discourse.nixos.org/t/purenix-nix-backend-for-purescript/15756/3).
+It explains how to get to the point where you can actually start writing
+PureScript and compiling it to Nix.
+
+Once you have your environment setup, you'll likely be interested in what
+libraries are available for PureNix.  The currently available libraries
+are in the [purenix-org](https://github.com/purenix-org/) organization on
+GitHub.  There are also two PureNix issues you may be interested in:
+
+-   A [tracking issue](https://github.com/purenix-org/purenix/issues/37)
+    for which PureScript standard libraries still need to be ported to PureNix.
+    If you're looking for a way to help out PureNix, this is a good place
+    to start!
+-   An issue for putting together a real
+    [PureNix package set](https://github.com/purenix-org/purenix/issues/36).
+    A PureScript/PureNix package set is similar to a Stackage resolver.  It is
+    used by Spago to make it easier to find library versions that work together.
+
+You may also be interested in [Pursuit](https://pursuit.purescript.org/).
+This is like Hoogle, but for PureScript.  Pursuit doesn't know anything
+about PureNix, but the API of the PureNix libraries that have been ported so
+far is almost the same as the PureScript version of the libraries, so searching
+for functions on Pursuit is still helpful.
+
+Now that you have an idea how to get started with PureNix, what sorts of
+developers will find PureNix easy to use?
+
+## Who is PureNix for?
+
+If you have experience with PureScript, Nix, and Haskell, PureNix will be easy
+to get started with.  You may have to write some FFI for Nix builtins, but
+other than that you should have no trouble using PureNix.  It should feel
+quite natural.
```

Comment:
I feel like this section is a little add odds with what this post seems to be -- a practical introduction to PureNix. This seems more at home in post 2? Or maybe it's worth moving post 2 up, as a non-technical introduction?

Reply:
That's a good point.  Maybe I'll take this section and just make it its own post.  I could probably expand it a little as well.

Reply:
I removed this section in 80aa5e6.

Reply:
I re-added this section as a new post in 69a0505.